lib/livebook_web/live/session_live.ex

```diff
@@ -565,6 +572,8 @@ defmodule LivebookWeb.SessionLive do
         _url,
         %{assigns: %{live_action: :catch_all}} = socket
       ) do
+    assert_policy!(socket, :read)
```

Comment:
Ah this is tricky. So here we handle relative links, but we generally shouldn't someone viewing shared notebook to open/import relative notebooks. At the same time we shouldn't raise, rather redirect back to the notebook root url and show an error message.

Reply:
hmm. It's not a simple, is it? ;(

Reply:
Actually a simple `if` should cover it. We can check the policy and either proceed or `redirect + put_flash`.

Reply:
Isn't this something weird? [code](https://github.com/livebook-dev/livebook/pull/1128/files#diff-255c2c5ca297bc981ca82b0e7a6957491bb000b1a0a265c2f43493b5e0c2c03bR581-R599)

Reply:
To clarify, this would happen if someone in ready-only mode clicked on a relative notebook link. It's a perfectly valid scenario, so we need to handle it gracefully.

We want to allow `:catch_all` for someone with `:read` access. Then, we still call `assert_live_action_access!`, but in that case it wouldn't raise.

So, basically we want to have a check like this

```elixir
if socket.assigns.policy.edit do do
  # regular stuff
else
  # put error "No access to navigate"
  # redirect to the notebook page (not the homepage), so that the path stays as expected
end
```

Reply:
Yeah, that's a little tricky